core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/RowParser.java

```diff
@@ -0,0 +1,360 @@
+package org.mobilitydata.gtfsvalidator.parsing;
+
+import org.apache.commons.validator.routines.EmailValidator;
+import org.apache.commons.validator.routines.UrlValidator;
+import org.mobilitydata.gtfsvalidator.notice.*;
+import org.mobilitydata.gtfsvalidator.type.GtfsColor;
+import org.mobilitydata.gtfsvalidator.type.GtfsDate;
+import org.mobilitydata.gtfsvalidator.type.GtfsTime;
+
+import java.time.ZoneId;
+import java.util.Currency;
+import java.util.Locale;
+import java.util.TimeZone;
+
+/**
+ * Parses cells of a CSV row as values of requested data types.
+ * <p>
+ * Interface functions of this class receive an instance of {@code NoticeContainer}. If a cell value cannot be parsed,
+ * these functions add a notice to the container, return null and don't throw an exception.
+ */
+public class RowParser {
+    public static final boolean REQUIRED = true;
+    public static final boolean OPTIONAL = false;
+    private final NoticeContainer noticeContainer;
+    private final ValueParser<Boolean> booleanParser = new ValueParser("boolean") {
+        @Override
+        Boolean parseString(String s) {
+            int i = Integer.parseInt(s);
+            switch (i) {
+                case 0:
+                    return false;
+                case 1:
+                    return true;
+                default:
+                    throw new IllegalArgumentException("Illegal boolean value" + i);
+            }
+        }
+    };
+    private final ValueParser<Integer> integerParser = new ValueParser("integer") {
+        @Override
+        Integer parseString(String s) {
+            return Integer.parseInt(s);
+        }
+    };
+    private final ValueParser<Double> floatParser = new ValueParser("float") {
+        @Override
+        Double parseString(String s) {
+            return Double.parseDouble(s);
+        }
+    };
+    private final ValueParser<TimeZone> timezoneParser = new ValueParser("timezone") {
+        @Override
+        TimeZone parseString(String s) {
+            return TimeZone.getTimeZone(ZoneId.of(s));
+        }
+    };
+    private final ValueParser<Locale> languageCodeParser = new ValueParser("language code") {
+        // TODO: Enhance checks for IETF BCP 47 language code.
```

Comment:
Just flagging the TODO here - we should implement in this PR or open an issue to finish it later.

Reply:
I propose to use the FIXME "code" instead of TODO.

Reply:
Done.